inc/BrainPadTemperatureSensor.h

```diff
@@ -2,15 +2,16 @@
 #define BPTEMPERATURESENSOR_H
 
 #include "BrainPadPin.h"
+#include "AnalogSensor.h"
 
 namespace codal {
-    class BrainPadTemperatureSensor {
-        BrainPadPin& pin;
+    class BrainPadTemperatureSensor : public codal::AnalogSensor {
+        codal::AnalogSensor* sensor = this;
```

Comment:
Why do we need this field? To access the base class's `getValue` just do `codal::AnalogSensor::getvalue` I believe